projects/fsastorefrontlib/src/cms-components/assets-table/resolve-asset-value.pipe.ts

```diff
@@ -0,0 +1,33 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { TranslationService } from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { RequestType } from '../../occ';
+
+@Pipe({ name: 'resolveAssetValue' })
+export class ResolveAssetValuePipe implements PipeTransform {
+  constructor(private translation: TranslationService) {}
+
+  transform(value: any, ...args: any[]): Observable<string> {
+    const asset = args[0];
+
+    if (value.propName) {
+      return of(
+        value.value.split('.').reduce((accum, curr) => accum?.[curr], asset)
+      );
+    }
+
+    return this.allowedFSRequestTypesIsClaim(asset)
+      ? this.translation.translate(value.value)
+      : this.translation.translate('fscommon.notAvailable');
+  }
+
+  allowedFSRequestTypesIsClaim(asset) {
+    const allowedFSRequestTypes = asset.categoryData.allowedFSRequestTypes;
+
+    return (
+      allowedFSRequestTypes
```

Comment:
Could this have been in a constant and perhaps use findIndex instead of map and indexOf?